eng/common/TestResources/remove-test-resources.yml

```diff
@@ -1,57 +1,41 @@
+# }
+# The Remove-TestResources.ps1 script accommodates extra parameters so it will
+# not error when parameters are provided which the script doesn't use.
 
 steps:
-  - pwsh: >
-      eng/common/TestResources/Remove-TestResources.ps1
-      -ResourceGroupName "${env:AZURE_RESOURCEGROUP_NAME}"
-      -TenantId '$(aad-azure-sdk-test-tenant-id)'
-      -SubscriptionId '$(azure-subscription-id)'
-      -ProvisionerApplicationId '$(aad-azure-sdk-test-client-id)'
-      -ProvisionerApplicationSecret '$(aad-azure-sdk-test-client-secret)'
-      -ServiceDirectory '${{ parameters.ServiceDirectory }}'
-      -Environment 'AzureCloud'
-      -Force
-      -Verbose
-    displayName: Remove test resources (AzureCloud)
-    condition: and(ne(variables['AZURE_RESOURCEGROUP_NAME'], ''), eq(variables['CloudType'], 'AzureCloud'))
-    continueOnError: true
+  # On MacOS there are problems with access to the ~/.Azure folder. This
+  # workaround fixes the problem.
+  - bash: sudo rm -rf ~/.Azure
+    condition: and(succeeded(), eq(variables['Agent.OS'], 'Darwin'))
+    displayName: Clean out ~/.Azure
 
-  - pwsh: >
-      eng/common/TestResources/Remove-TestResources.ps1
-      -ResourceGroupName "${env:AZURE_RESOURCEGROUP_NAME}"
-      -TenantId '$(aad-azure-sdk-test-tenant-id-gov)'
-      -SubscriptionId '$(azure-subscription-id-gov)'
-      -ProvisionerApplicationId '$(aad-azure-sdk-test-client-id-gov)'
-      -ProvisionerApplicationSecret '$(aad-azure-sdk-test-client-secret-gov)'
-      -ServiceDirectory '${{ parameters.ServiceDirectory }}'
-      -Environment 'AzureUSGovernment'
-      -Force
-      -Verbose
-    displayName: Remove test resources (AzureUSGovernment)
-    condition: and(ne(variables['AZURE_RESOURCEGROUP_NAME'], ''), eq(variables['CloudType'], 'AzureUSGovernment'))
-    continueOnError: true
+  - pwsh: |
+      $subscriptionConfiguration = @"
+        ${{ parameters.SubscriptionConfiguration }}
+      "@ | ConvertFrom-Json -AsHashtable;
 
-  - pwsh: >
-      eng/common/TestResources/Remove-TestResources.ps1
-      -ResourceGroupName "${env:AZURE_RESOURCEGROUP_NAME}"
-      -TenantId '$(aad-azure-sdk-test-tenant-id-cn)'
-      -SubscriptionId '$(azure-subscription-id-cn)'
-      -ProvisionerApplicationId '$(aad-azure-sdk-test-client-id-cn)'
-      -ProvisionerApplicationSecret '$(aad-azure-sdk-test-client-secret-cn)'
-      -ServiceDirectory '${{ parameters.ServiceDirectory }}'
-      -Environment 'AzureChinaCloud'
-      -Force
-      -Verbose
-    displayName: Remove test resources (AzureChinaCloud)
-    condition: and(ne(variables['AZURE_RESOURCEGROUP_NAME'], ''), eq(variables['CloudType'], 'AzureChinaCloud'))
+      &eng/common/TestResources/Remove-TestResources.ps1 `
+        -ResourceGroupName "${env:AZURE_RESOURCEGROUP_NAME}" `
+        -ServiceDirectory ${{ parameters.ServiceDirectory }} `
+        @subscriptionConfiguration `
+        -Force `
+        -Verbose
+    displayName: Remove test resources
     continueOnError: true
```

Comment:
Not related to specific changes but I'm curious why we have continueOnError set to true. Do we expect this step to fail sometimes?

Reply:
If it does, it's not a big deal since the clean-up script will delete within 24hrs. No reason to fail the run if we can't clean up resources at the end, right?

Reply:
Correct. There's no reason to block if there's a (sometimes transient) failure in resource removal.